algorithms/src/Kokkos_Random.hpp

```diff
@@ -903,24 +903,46 @@ class Random_XorShift64_Pool {
 #else
   Random_XorShift64_Pool()   = default;
 #endif
+
   Random_XorShift64_Pool(uint64_t seed) {
-    num_states_ = 0;
+    init(execution_space(), seed, execution_space().concurrency());
+    execution_space().fence("Random_XorShift64_Pool: Constructor");
+  }
```

Comment:
Could delegate to the other constructor

Reply:
Doesn't really safe any characters, and does add one more layer of instantiation.

Reply:
I would also prefer constructor delegation, but it is not critical and there is definitely too many cook in this kitchen/PR :)